### PR TITLE
Replace `eslint` with `xo`

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Various rules have been disabled for now, because the TypeScript code is about to be replaced soon anyway.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -34,21 +32,21 @@ interface IResult {
...
   output.appendLine('[Remark]')
-  output.append(msg)
+  output.append(message)
   output.show()
 }
 
-function getPlugins(list: string[]): Promise<IPlugin[]> {
-  const root = vscode.workspace.rootPath || ''
+async function getPlugins(list: string[]): Promise<IPlugin[]> {
+  const root = vscode.workspace.rootPath ?? ''
```
Comment (wooorm) [inline src/extension.ts:49]: VS Code is Electron right, or not? Will it support such new features?
Should this XO rule be turned off?

File: src/extension.ts [hunk 2]
```diff
@@ -232,14 +230,13 @@ export function activate(context: vscode.ExtensionContext) {
             )
 
             return action
-          } catch (error) {
+          } catch (error: any) {
```
Comment (wooorm) [inline src/extension.ts:233]: `unknown` and `as Error`?

File: src/test/suite/index.ts [hunk 3]
```diff
@@ -33,16 +32,16 @@ export async function run(): Promise<void> {
   })
   const testsRoot = path.resolve(__dirname, '..')
   const files = glob.sync('**/*.test.js', {cwd: testsRoot})
-  files.forEach((f) => mocha.addFile(path.resolve(testsRoot, f)))
+  for (const f of files) mocha.addFile(path.resolve(testsRoot, f))
```
Comment (wooorm) [inline src/test/suite/index.ts:35]: Probably a good time to call this `file` instead

File: src/test/suite/workspace.test.ts [hunk 4]
```diff
@@ -1,24 +1,21 @@
 'use strict'
 
+import * as path from 'path'
```
Comment (wooorm) [inline src/test/suite/workspace.test.ts:3]: I believe most of these `* as x` should just be `import path from 'path'` — the stars are not needed, they work in some libraries, but not all

**Comment by wooorm** [comment]
Also: build error

File: src/test/suite/workspace.test.ts (hunk 4 above)
Comment (remcohaszing) [reply src/test/suite/workspace.test.ts:3]: What you’re saying it true for ESM, but not for TypeScript that’s compiled to CJS, which is what this is.

File: src/extension.ts (hunk 2 above)
Comment (remcohaszing) [reply src/extension.ts:233]: I’d rather not. These types were wrong and this PR changes nothing about that. It’s going to be fully replaced soon. I’m purely trying to make the diffs manageable by doing some chores that can be done before replacing the entire TypeScript code with the language server implementation.

File: src/extension.ts (hunk 1 above)
Comment (remcohaszing) [reply src/extension.ts:49]: This is transpiled by TypeScript.

**Comment by wooorm** [comment]
missed the replacing part

**Review by wooorm — APPROVED** [review]

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**